extensions/mongodb/src/test/java/com/hazelcast/jet/mongodb/datalink/MongoDataLinkTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.jet.mongodb.datalink;
+
+import com.hazelcast.datalink.DataLinkResource;
+import com.hazelcast.jet.mongodb.AbstractMongoDBTest;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoDatabase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+
+import static com.hazelcast.jet.mongodb.datalink.MongoDataLink.mongoDataLinkConf;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class MongoDataLinkTest extends AbstractMongoDBTest {
+    private MongoDataLink dataLink;
+    private String connectionString;
+
+    @Before
+    public void setup() {
+        this.connectionString = mongoContainer.getConnectionString();
+    }
+
+    @After
+    public void cleanup() {
+        dataLink.destroy();
+        dataLink = null;
+    }
+
+    @Test
+    public void should_return_same_link() {
+        dataLink = new MongoDataLink(mongoDataLinkConf("mongo", connectionString));
+
+        MongoClient client1 = dataLink.getClient();
+        MongoClient client2 = dataLink.getClient();
+
+        assertThat(client1).isNotNull();
+        assertThat(client2).isNotNull();
+        assertThat(client1).isSameAs(client2);
```

Comment:
Can you check that dataLink.ref counter is still the same at the end of the test

Reply:
ReferenceCounter is an implementation details; here I'm looking at results of RC's work only (black boxed)